src/WebJobs.Script/Workers/Rpc/RpcWorkerConcurrencyManager.cs

```diff
@@ -0,0 +1,243 @@
+
+            // Check if there are initializing language workers
+            int notInitializedWorkersCount = workerChannels.Count() - initializedWorkers.Count();
+
+            if (notInitializedWorkersCount == 0)
+            {
+                // Check how many channels are oveloaded
+                List<WorkerDescription> descriptions = new List<WorkerDescription>();
+                foreach (IRpcWorkerChannel channel in initializedWorkers)
+                {
+                    if (channel is IRpcWorkerConcurrencyChannel concurrencyChannel)
+                    {
+                        WorkerStatus workerStatus = concurrencyChannel.GetWorkerStatus();
+                        bool overloaded = IsOverloaded(workerStatus);
+                        descriptions.Add(new WorkerDescription()
+                        {
+                            Channel = channel,
+                            WorkerStatus = workerStatus,
+                            Overloaded = overloaded
+                        });
+                    }
+                }
+
+                int overloadedCount = descriptions.Where(x => x.Overloaded == true).Count();
+                if (overloadedCount > 0)
+                {
+                    if (initializedWorkers.Count() < _concurrencyOptions.Value.MaxWorkerCount)
+                    {
+                        _logger.LogDebug($"Adding a new worker, overloaded workers = {overloadedCount}, initialized workers = {initializedWorkers.Count()} ");
+                        result = true;
+                    }
+                }
+
+                if (result == true || _logStateStopWatch.Elapsed > _logStateInterval)
+                {
+                    StringBuilder sb = new StringBuilder();
+                    foreach (WorkerDescription description in descriptions)
+                    {
+                        sb.Append(LogWorkerState(description));
+                        sb.Append(Environment.NewLine);
+                    }
+                    _logStateStopWatch.Restart();
+                    _logger.LogDebug(sb.ToString());
+                }
+            }
+
+            return result;
+        }
+
+        internal bool IsOverloaded(WorkerStatus status)
+        {
+            if (status.RpcWorkerStats.LatencyHistory.Count() >= _concurrencyOptions.Value.HistorySize)
+            {
+                int overloadedCount = status.RpcWorkerStats.LatencyHistory.Where(x => x.TotalMilliseconds >= _concurrencyOptions.Value.LatencyThreshold.TotalMilliseconds).Count();
+                double proportion = (double)overloadedCount / _concurrencyOptions.Value.HistorySize;
+
+                return proportion >= _concurrencyOptions.Value.HistoryThreshold;
+            }
+            return false;
+        }
+
+        internal string LogWorkerState(WorkerDescription desc)
+        {
+            string formattedLoadHistory = string.Empty, formattedLatencyHistory = string.Empty;
```

Comment:
As mentioned above, all this process stats logging should be removed. It'll already be logged by the WebJobs SDK HostProcessMonitor peridically.

Reply:
I will update the PR once #7406 is merged.

Reply:
This is updated.